lib/attachments.js

```diff
@@ -1,72 +1,16 @@
-const client = require('elasticio-rest-node')();
-const http = require('http');
-const url = require('url');
-
-exports.addAttachment = addAttachment;
-
-function addAttachment(msg, name, stream, contentLength) {
-    console.log("About to upload attachment stream of length %s", contentLength);
-
-    return client.resources.storage
-        .createSignedUrl()
-        .then(onSignedUrl);
-
-    function onSignedUrl(result) {
-        console.log("Created signed URL=%s", result.get_url);
-
-        const opts = createRequestOptions(contentLength, result.put_url);
-
-        return uploadFile(stream, opts)
-            .then(addUrlAttachment.bind(null, msg, name, result.get_url, contentLength));
-    }
+const { AttachmentProcessor } = require('@elastic.io/component-commons-library');
+
+async function addAttachment(msg, name, stream, contentLength) {
+  // eslint-disable-next-line no-return-await
+  return await new AttachmentProcessor().uploadAttachment(stream, 'stream')
```

Comment:
Refactor this to try/catch with await logic. Also remove `eslint-disable-next-line no-return-await`